Assets/Scripts/GameStructure/NarrativeSystem.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            _timeUntilTrigger -= Time.deltaTime;
+
+            if (_timeUntilTrigger <= 0f)
+            {
+                MoveToNextPage(); // A page consists of panels.Length # of panels
+            }
+        }
+
+        private void ShowFirstPanel()
+        {
+            backButton.interactable = false;
+            ShowPage(0);
+        }
+
+        private void ShowPage(int pageOfPanels)
+        {
+            var totalPanelsBeforeThisPage = _panelsPerPage.Take(pageOfPanels).Sum();
+
+            if (_panelsPerPage[pageOfPanels] == 1)
+            {
+                // Show single panel only
+                fullScreenPanel.sprite = _thisNarrative.allPanels[totalPanelsBeforeThisPage].panelImage;
+                fullScreenPanel.gameObject.SetActive(true);
+                foreach (var p in panels) { p.gameObject.SetActive(false); }
+            }
+            else // We will show as many panels are we show on screen
+            {
+                fullScreenPanel.gameObject.SetActive(false);
+                for (var i = 0; i < panels.Length; i++)
+                {
+                    var thisPanelSprite = _thisNarrative.allPanels[totalPanelsBeforeThisPage + i].panelImage;
+                    panels[i].sprite = thisPanelSprite;
+                    panels[i].gameObject.SetActive(true);
+                }
+            }
+
+            _pageOnScreen = pageOfPanels;
+        }
+
+        private void ResetTimer()
+        {
+            _timeUntilTrigger = timeToMoveToNextPanel;
+        }
+
+        public void MoveToNextPage()
+        {
+            ResetTimer();
+
+            // move to next page of panels or end narrative
+            if (_pageOnScreen + 1 >= _panelsPerPage.Count)
+            {
+                EndNarrative();
+            }
+            else
+            {
+                ShowPage(_pageOnScreen + 1);
+            }
+
+            backButton.interactable = true;
+        }
+
+        public void EndNarrative()
+        {
+            _actionOnEnd?.Invoke(this); // could consider event triggers also but this seems ok for now
+        }
+
+        public void MoveToPreviousPage()
+        {
+            ResetTimer();
+
+            // move to prev page of panels if it exists
+            if (_pageOnScreen > 0)
+            {
+                ShowPage(_pageOnScreen - 1);
+                if (_pageOnScreen == 0)
+                {
+                    backButton.interactable = false;
+                }
+            }
+        }
+
+        private void OnUserInteraction()
+        {
+            ResetTimer();
+        }
+
+        private void OnEnable()
+        {
+            // Populate data for the system from the narrative manager
+            _dataManagerRef = NarrativeDataManager.Instance;
+
+            foreach (var button in allButtonsToPauseAutoScroll)
+            {
+                button.onClick.AddListener(OnUserInteraction);
+            }
+        }
+
+        private void OnDisable()
```

Comment:
Should this invoke ResetTimer()?

Reply:
Nah it's only used inside this class and only checked and updated in update()